pxtcompiler/emitter/service.ts

```diff
@@ -1970,6 +1971,83 @@ namespace ts.pxtc.service {
                 return false
             }
 
+            function snippetFromSpriteEditorParams(opts: pxt.Map<string>): SnippetNode | null {
+                // TODO: Generalize this to share implementation with FieldSpriteEditor in field_sprite.ts
+                const parsed = {
+                    initColor: 0,
+                    initWidth: 16,
+                    initHeight: 16,
+                };
+
+                if (opts?.sizes) {
+                    const pairs = opts.sizes.split(";");
+                    const sizes: [number, number][] = [];
+                    for (let i = 0; i < pairs.length; i++) {
+                        const pair = pairs[i].split(",");
+                        if (pair.length !== 2) {
+                            continue;
+                        }
+
+                        let width = parseInt(pair[0]);
+                        let height = parseInt(pair[1]);
+
+                        if (isNaN(width) || isNaN(height)) {
+                            continue;
+                        }
+
+                        const screenSize = runtimeOps?.screenSize;
+                        if (width < 0 && screenSize)
+                            width = screenSize.width;
+                        if (height < 0 && screenSize)
+                            height = screenSize.height;
+
+                        sizes.push([width, height]);
+                    }
+                    if (sizes.length > 0) {
+                        parsed.initWidth = sizes[0][0];
+                        parsed.initHeight = sizes[0][1];
+                    }
+                }
+
+                parsed.initColor = withDefault(opts?.initColor, parsed.initColor);
+                parsed.initWidth = withDefault(opts?.initWidth, parsed.initWidth);
+                parsed.initHeight = withDefault(opts?.initHeight, parsed.initHeight);
+
+                return pxt.sprite.imageLiteralFromDimensions(parsed.initWidth, parsed.initHeight, parsed.initColor, fileType);
+
+                function withDefault(raw: string, def: number) {
+                    const res = parseInt(raw);
+                    if (isNaN(res)) {
+                        return def;
+                    }
+                    return res;
+                }
+            }
+
+            function getDefaultValueFromFieldEditor(paramName: string): SnippetNode | null {
+                const blockParam = attrs._def?.parameters?.find((p) => p.name === name);
```

Comment:
Don't reference _def directly, in pxtlib we have a function called `pxt.blocks.compileInfo` that parses it